ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/views/CreateSupportTicketView.kt

```diff
@@ -0,0 +1,328 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Snackbar
+import androidx.compose.material3.SnackbarHost
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.revenuecat.purchases.customercenter.CustomerCenterConfigData
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.CustomerCenterConstants.Layout.SECTION_SPACING
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.CustomerCenterConstants.Layout.SECTION_TITLE_BOTTOM_PADDING
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.composables.SettingsButton
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.composables.SettingsButtonConfig
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CreateSupportTicketData
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.data.CustomerCenterConfigTestData
+import com.revenuecat.purchases.ui.revenuecatui.customercenter.theme.CustomerCenterPreviewTheme
+
+private const val MAX_DESCRIPTION_LENGTH = 250
+
+private fun isValidEmail(email: String): Boolean {
+    return email.isNotBlank() && Patterns.EMAIL_ADDRESS.matcher(email).matches()
+}
+
+private data class EmailInputState(
+    val email: String,
+    val onEmailChange: (String) -> Unit,
+    val onFocusChanged: (Boolean) -> Unit,
+    val showError: Boolean,
+    val enabled: Boolean,
+)
+
+private data class DescriptionInputState(
+    val description: String,
+    val onDescriptionChange: (String) -> Unit,
+    val enabled: Boolean,
+)
+
+@JvmSynthetic
+@Composable
+internal fun CreateSupportTicketView(
+    data: CreateSupportTicketData,
+    localization: CustomerCenterConfigData.Localization,
+    modifier: Modifier = Modifier,
+) {
+    var email by remember { mutableStateOf("") }
+    var emailTouched by remember { mutableStateOf(false) }
```

Comment:
Do we need this? or we can check if `email != ""` instead? Also maybe `emailDirty` fits better, so it doesn't seem like it means that the email field was touched with the finger

Reply:
The reason I added this was if I just checked if it was empty, then it would show the validation error from the start, which didn't feel right to me (but happy to change it to be that way)

I will update the var name so it's clearer